src/addons/transitions/ReactCSSTransitionGroupChild.js

```diff
@@ -77,6 +79,10 @@ var ReactCSSTransitionGroupChild = React.createClass({
       CSSCore.removeClass(node, className);
       CSSCore.removeClass(node, activeClassName);
 
+      if (childRef.componentDidTransition) {
+        childRef.componentDidTransition(animationType);
```

Comment:
@jmeas That's correct. `componentWillTransition` is called just before the transition begins and `componentDidTransition` is called just after it ends. The order looks backwards because of the position of endListener in the source code.